### DISCUSSION TITLE
URL Params from OAuth Redirect

### DISCUSSION THREAD

**Discussion by NewBeeSD** [supabase/supabase#21309] [Questions]
I am trying to create a FastAPI backend website. So here is something I am stuck on. For Google OAuth sign in feature... I set the redirect URL to **https://mydomain.com/authredirect**. Now when I am testing it by going to http://<my project id>/auth/v1/authorize?provider=google  then it is redirecting it to **https://mydomain.com/authredirect#access_token=<token1>&refresh_token=<token2>&token_type=bearer**. My plan was to create a function in FastAPI backend like

@app.get("/authredirect")
async def auth_redirect(request: Request) 
    query_params = request.query_params
    print(query_params)
    
--> this query_params is to read all the token1 and token2 from this part...  #access_token=<token1>&refresh_token=<token2>&token_type=bearer and get user information for the logged in user.

But because this part of the url is after # -> I cannot read them as part of url parameter/value. 

Question is: Is my approach the right approach? How do I get the logged-in user information ?

**Comment by charislam** [comment]
To implement Auth on a server, you'll need to use a [PKCE flow](https://supabase.com/docs/guides/auth/server-side/oauth-with-pkce-flow-for-ssr), which sends a code in the query param (which _is_ accessible on the server) instead of directly putting the access token in the hash. You can then exchange the code for an access token.

**Reply by RobertoValente** [reply]
Hi charislam, can you check my logic/code? Of course that i'm doing something wrong but i didn't figure it out...
![Screenshot_1](https://github.com/supabase/supabase/assets/73560955/80bc450b-3514-4e11-a444-51e3d1c17bd9)

**Reply by J0** [reply]
Hey @RobertoValente,

Thanks for the query! Do you mind sharing how you're initialising the Spuabase Client?

**Reply by RobertoValente** [reply]
I've already tried few ways, and here we go:
```js
const supabase = createClient('URL', 'KEY');

const supabase = createClient('URL', 'KEY', auth: {autoRefreshToken: true, detectSessionInUrl: true, persistSession: true});
```

- I've also changed some boolean values. Btw, I'm not sure of the syntax cause right now, I'm not able to check my code.
